components/datetime/src/error.rs

```diff
@@ -52,6 +53,12 @@ pub enum DateTimeFormatError {
     /// An error originating from a missing era symbol in the data.
     #[displaydoc("Data file missing era symbol for era code {0}")]
     MissingEraSymbol(TinyStr16),
+    /// An error from mixing calendar types
+    #[displaydoc("AnyDateTimeFormat for {0} calendar was given a {0:?} calendar")]
+    MismatchedAnyCalendar(AnyCalendarKind, Option<AnyCalendarKind>),
+    /// An error from attempting to build AnyDateTimeFormat without a calendar in the locale
+    #[displaydoc("AnyDateTimeFormat requires locale to have a known u-ca- calendar subtag")]
+    MissingCalendarOnLocale,
```

Comment:
(this error should never happen)

Reply:
decision: default to gregory for now (ideally, ISO but we don't have ISO data), wait for vertical fallback, leave giant scary PLZ NO SHIP comment